tests/drivers/random/randomnumgen_basic_api/testcase.yaml

```diff
@@ -0,0 +1,4 @@
+tests:
+-   test_random_num_genrator:
+        platform_whitelist: esp32
```

Comment:
instead of a whitelist we should do something like:

filter: CONFIG_RANDOM_HAS_DRIVER